x-pack/plugins/ml/public/embeddables/single_metric_viewer/single_metric_viewer_embeddable_factory.tsx

```diff
@@ -0,0 +1,438 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { StartServicesAccessor } from '@kbn/core/public';
+import { KibanaContextProvider } from '@kbn/kibana-react-plugin/public';
+import { KibanaThemeProvider } from '@kbn/react-kibana-context-theme';
+import { pick } from 'lodash';
+
+import type { ReactEmbeddableFactory } from '@kbn/embeddable-plugin/public';
+import type { TimeRange } from '@kbn/es-query';
+import { EuiResizeObserver } from '@elastic/eui';
+import React, { useCallback, useEffect, useRef, useState, Suspense } from 'react';
+import useUnmount from 'react-use/lib/useUnmount';
+import moment from 'moment';
+import {
+  apiHasParentApi,
+  apiPublishesTimeRange,
+  initializeTimeRange,
+  initializeTitles,
+  useStateFromPublishingSubject,
+} from '@kbn/presentation-publishing';
+import { DatePickerContextProvider, type DatePickerDependencies } from '@kbn/ml-date-picker';
+import { UI_SETTINGS } from '@kbn/data-plugin/common';
+import type { MlJob } from '@elastic/elasticsearch/lib/api/types';
+import usePrevious from 'react-use/lib/usePrevious';
+import type { Observable } from 'rxjs';
+import { throttle } from 'lodash';
+import { BehaviorSubject, combineLatest, map, of, Subscription } from 'rxjs';
+import { ANOMALY_SINGLE_METRIC_VIEWER_EMBEDDABLE_TYPE } from '..';
+import type { MlDependencies } from '../../application/app';
+import { HttpService } from '../../application/services/http_service';
+import { AnomalyExplorerChartsService } from '../../application/services/anomaly_explorer_charts_service';
+import type { MlPluginStart, MlStartDependencies } from '../../plugin';
+import { EmbeddableLoading } from '../common/components/embeddable_loading_fallback';
+import type {
+  SingleMetricViewerEmbeddableServices,
+  SingleMetricViewerEmbeddableApi,
+  SingleMetricViewerEmbeddableState,
+} from '../types';
+import { initializeSingleMetricViewerControls } from './single_metric_viewer_controls_initializer';
+import { initializeSingleMetricViewerDataFetcher } from './single_metric_viewer_data_fetcher';
+import { TimeSeriesExplorerEmbeddableChart } from '../../application/timeseriesexplorer/timeseriesexplorer_embeddable_chart';
+import { APP_STATE_ACTION } from '../../application/timeseriesexplorer/timeseriesexplorer_constants';
+import './_index.scss';
+
+const RESIZE_THROTTLE_TIME_MS = 500;
+const containerPadding = 10;
+interface AppStateZoom {
+  from?: string;
+  to?: string;
+}
+
+/**
+ * Provides the services required by the Anomaly Swimlane Embeddable.
+ */
+export const getServices = async (
```

Comment:
nit: this file is getting a little bit long, the services could be a good place to split off. Additionally, I wonder if there's any commonality with this and the swimlane's services?

Reply:
Split off in fbb2f38310a6e1b210b1f67c28d0a5958b331034